src/main/java/tigase/vhosts/VHostJDBCRepository.java

```diff
@@ -110,9 +110,9 @@ public VHostItem getItemInstance() {
 		VHostItemImpl item = VHostRepoDefaults.getItemInstance();
 		item.setExtensionManager(extensionManager);
 		VHostItemImpl.VHostItemWrapper wrapper = new VHostItemImpl.VHostItemWrapper();
-		wrapper.setItem(item);
-		wrapper.setDefaults(defaults);
-		wrapper.setVHostDefaults(this.vhostDefaults);
+		wrapper.setItem(item, false);
+//		wrapper.setDefaultVHost(defaultVHost, false);  // we shouldn't set default in bare-boned instance
```

Comment:
At first I agreed with you that this would be an issue, but trying to address that I came to realisation that (apart that current "defaults" is somewhat incorrect) I'm not sure we want to present those default settings to the user during creation!

This wasn't specified in any place, but from what I remember the idea was mostly like this: if value is not set (empty/null/false) then use default (from `default` vhost), otherwise use value configured for the VHost (and later on, determine 'effective' value of the extension). 

Of course the best way would be to have explicit configuration what is taken from default and what is configured, but that would increase complexity…

Reply:
Well, I agree but I do think that use should see `effective` settings during creation of vhost, ie. if installation forces TLS requirement it should be visible during creation and not possible to be changed, etc.

Why would someone want to see something which will not be taken into account as defaults will override it?

If I recall, you asked me not so long ago to show user during modifications/creation effective settings as in other case users would see values which could be overridden by defaults and would report issues that "settings do not work".

I'm pretty sure that we should present the user with effective values all the time.

Reply:
OK, I think we have a couple of issues here. It seems that we mix two concepts: default configuration (that could override configuration system-wise, as in the case of required TLS, and in case of that particular option it should stay that way) and "default" configuration, that serves as a fallback if no configuration is done for particular VHost (so, in order: particular vhost configuration -> `default` VHost configuration -> global system defaults). Above solution fulfils the second pattern, though, not without issues. It assumes that lack of value equals default configuration, instead of presenting user with actual, effective value.

I'd propose, for now, adding instruction message/warning, that some configuration options may be influenced by configuration set in `default` VHost.

In the future, we can address the problem properly with tri-state fields, that will help make it explicit.

What do you think?

Reply:
I think that solution with additional warning is acceptable. 

However, I'm wondering if we actually have an issue with default settings. As far as I remember, we had global settings which were used as a way to enforce some values for vhosts. Later on, we have provided a way to set global values as "defaults" so vhosts would use them if not set.

Right now, I wonder, if we should not get rid of those defaults and only leave a way of settings that should be enforced. Rest would be used from hardcoded defaults. Removing those "defaults" would simplify workflow. If I recall the only reason we added defaults was to make migration easier and to not add new required fields to adhoc command form. If we would leave only hardcoded defaults, that should work in the same way and would simplify workflow. But that is something to consider in the future.

Reply:
Thank you for your comments. I comited instructions with clarification and added issue for future versions: https://projects.tigase.net/issue/server-1155